activerecord/lib/active_record/aggregations.rb

```diff
@@ -264,7 +264,7 @@ def writer_method(name, class_name, mapping, allow_nil, converter)
               end
 
               hash_from_multiparameter_assignment = part.is_a?(Hash) &&
-                part.each_key.all? { |k| k.is_a?(Integer) }
+                part.keys.all?(Integer)
```

Comment:
Just curious why changed `each_key` to `keys`?

Reply:
I compared both and `#keys` seemed to work better:

```
Warming up --------------------------------------
               #keys   367.000  i/100ms
           #each_key   267.000  i/100ms
Calculating -------------------------------------
               #keys      3.354k (± 9.8%) i/s -     16.882k in   5.084377s
           #each_key      2.434k (± 6.8%) i/s -     12.282k in   5.069831s

Comparison:
               #keys:     3354.0 i/s
           #each_key:     2434.2 i/s - 1.38x  (± 0.00) slower
```

```ruby
require "benchmark/ips"

hsh = (1..10000).inject({}) { |h, n| h[n] = n; h; }

Benchmark.ips do |x|
  x.report("#keys") do
    hsh.keys.all?(Integer)
  end

  x.report("#each_key") do
    hsh.each_key.all?(Integer)
  end

  x.compare!
end
```

Not a significant difference though, I can revert it.